src/libs/swapAndBridge/swapAndBridge.ts

```diff
@@ -84,15 +85,58 @@ const getActiveRoutesUpdateInterval = (minServiceTime?: number) => {
   return 15000
 }
 
-const buildSwapAndBridgeUserRequests = (
+const buildRevokeApprovalIfNeeded = async (
+  userTx: SocketAPISendTransactionRequest,
+  account: Account,
+  provider: RPCProvider
+): Promise<Call | undefined> => {
+  if (!userTx.approvalData) return
+  const erc20Contract = new Contract(userTx.approvalData.approvalTokenAddress, ERC20.abi, provider)
+  const requiredAmount = isSmartAccount(account)
+    ? BigInt(userTx.approvalData.minimumApprovalAmount)
+    : MaxUint256
+  const approveCallData = erc20Contract.interface.encodeFunctionData('approve', [
+    userTx.approvalData.allowanceTarget,
+    requiredAmount
+  ])
+
+  let fails = false
+  try {
+    await provider.call({
+      from: account.addr,
+      to: userTx.approvalData.approvalTokenAddress,
+      data: approveCallData
+    })
+  } catch (e) {
+    fails = true
+  }
+
+  if (!fails) return
+
+  return {
+    to: userTx.approvalData.approvalTokenAddress,
+    value: BigInt('0'),
+    data: erc20Contract.interface.encodeFunctionData('approve', [
+      userTx.approvalData.allowanceTarget,
+      BigInt(0)
+    ])
+  }
+}
+
+const buildSwapAndBridgeUserRequests = async (
   userTx: SocketAPISendTransactionRequest,
   networkId: string,
-  account: Account
+  account: Account,
+  provider: RPCProvider
 ) => {
   if (isSmartAccount(account)) {
```

Comment:
Not a bug.  
In the smart account case, I see we don't' check allowance. I think this is better as we're going to force an exact approval, which means the allowance will get reduced to 0 after the transaction succeeds. From a security perspective, this is better.

But I also imagine getting complains "why do I have to approve again, I already have an infinite approval for this token".

Imo, as smart accounts we should force allowances to go to 0. Maybe it will be a good idea to communicate this and see what other people think as well?